### ISSUE TITLE
Allow running Scala Steward as GitHub Actions step

### ISSUE THREAD

**Issue by lustefaniak** [fthomas/scala-steward#1319]
That would reduce visibility of how many projects are updated by Steward, but would simplify using it in private repositories.

**Comment by fthomas** [comment]
Could you elaborate what kind of changes this entails? Would Scala Steward need to use a new GitHub API? Is this similar to GitHub Apps? Where would Scala Steward run? On GitHub servers or on your own server for private repos?

**Comment by onesupercoder** [comment]
Github Actions has a free to use runner(currently bound to listening to 1 repo per runner) you can host on your own server and offers a paid plan to use their runners charged by minutes.  You can configure your repo to use certain runners.  Using a .github/workflows/[workflow-name].yml

https://help.github.com/en/actions/building-actions/about-actions. - according to the documentation it there are Docker container actions and Javascript Actions.

Type | Operating system
-- | --
Docker container | Linux
JavaScript | Linux, MacOS, Windows

I was just searching for a scala steward github action.

**Comment by alejandrohdezma** [comment]
@fthomas I've created a [Scala Steward Github Action](https://github.com/marketplace/actions/scala-steward-github-action). It has been placed on a separate repository but I will gladly contribute it to this repository or one under [scala-steward-org](https://github.com/scala-steward-org/) if you prefer it that way 😸 

With this action, anyone can add a new workflow that allows launching Scala Steward on any repository by just adding the following as `.github/workflows/scala-steward.yml`:

```
on:
  schedule:    
    - cron:  '0 0 * * 0'

jobs:
  scala-steward:
    runs-on: ubuntu-latest
    steps:
      - uses: alejandrohdezma/scala-steward-action@v1
        with:
          github-token: ${{ secrets.ADMIN_GITHUB_TOKEN }}
          gpg-secret-key: ${{ secrets.GPG_SCALA_STEWARD }}
```

An example of this Github Action being run on a repository can be found [here](https://github.com/alejandrohdezma/sbt-github/runs/635724554?check_suite_focus=true).

**Comment by fthomas** [comment]
This is very interesting, @alejandrohdezma! If you're OK with having it under the scala-steward-org organization, that would be amazing. I'll invite you to the organization, so that you can move it there.

One thing that is unclear to me is if Scala Steward as GitHub Action inevitably means that one Scala Steward instance works only on one repository. One benefit of having one instance running for 1000+ repos is that it only checks for updates of libraries that are used by `n` repos once in a configured time interval instead of `n` times. So instead of checking for sbt updates a 1000+ times it does it only once every few hours. It would be great if we wouldn't loose this optimization with GitHub Actions.

**Comment by alejandrohdezma** [comment]
The action should be located here now [scala-steward-org/scala-steward-action](https://github.com/scala-steward-org/scala-steward-action).

Regarding the optimization, maybe we can close this issue and move the conversation as a new issue in the action's repository. What do you think?

**Comment by fthomas** [comment]
Yeah, that makes sense. I'll open an issue in the action's repository.
